### PR TITLE
Give warning when a malformed JSON is dropped onto HiGlass

### LINKED ISSUE #145: Give warning when a malformed JSON is dropped onto higlass
Currently the only indication that the added JSON is not valid is found in the command line.

### PR DESCRIPTION
**Pull request by sehilyi**
## Description

> What was changed in this pull request?

Add a panel in the viewConfig editor view that shows whether the edited json code is valid according to the definition of the higlass json schema. The failure of parsing JSON codes gives an error message and invalid configs according to the schema definition gives warning messages.

> Why is it necessary?

People can easily check whether their viewConfig is valid or not without seeing the console messages in the browser.

schema-warning

Fixes #145 

## Checklist

- [ ] Set proper GitHub labels (e.g. v1.6+, ignore if you don't know)
- [ ] Unit tests added or updated
- [ ] Documentation added or updated
- [ ] Example(s) added or updated
- [ ] Update schema.json if there are changes to the viewconf JSON structure format
- [x] Screenshot for visual changes (e.g. new tracks or UI changes)
- [x] Updated CHANGELOG.md

### CODE REVIEW

File: app/scripts/ViewConfigEditor.js [hunk 1]
```diff
@@ -94,7 +128,35 @@ class ViewConfigEditor extends React.Component {
...
+    if (this.state.showLog) {
+      this.hideLog();
+    } else {
+      this.showLog();
+    }
+  }
+
   render() {
+    const logMessages = this.state.logMsgs.map((d, i) => {
+      const key = `${i}-${d.type}`;
```
Comment (pkerpedjiev) [inline app/scripts/ViewConfigEditor.js:149]: These keys may not be unique.

For example, you do something and get  one  error message `1-Warning`. Then you fix that error  and get a different warning which would now also be `1-Warning` but may have a different `msg`.

**Comment by pkerpedjiev** [comment]
Oh, would you mind fixing the build before merging?

**Comment by sehilyi** [comment]
I was testing with a monaco-editor (https://github.com/react-monaco-editor/react-monaco-editor) which I found it to be useful in editing higlass view configs. Some advantages:
- You can see warnings and errors directly on the code with yellow and red underlines and tooltip messages based on a JSON schema.
- Suggestions on typing object keys + auto complete.
- Collapsing and expanding brackets.

monaco-editor

If you think this can be useful in addition to showing the log messages, I can try adding this editor by replacing the current 'react-simple-code-editor'.

File: app/scripts/ViewConfigEditor.js (hunk 1 above)
Comment (sehilyi) [reply app/scripts/ViewConfigEditor.js]: You are right, and thanks for the explanation! I've replaced it to `d.msg`.

**Comment by pkerpedjiev** [comment]
About the Monaco editor, the higlass bundle size is already very large and since the viewconf editor is meant for quick one-off edits if the Monaco editor bundle is significantly larger than the current editor's bundle, I would vote for sticking with the slimmer version.

**Comment by sehilyi** [comment]
That is true, and it was my main concern with the Monaco editor as well. I will examine the size whether it is actually feasible to adopt it in higlass.

**Comment by sehilyi** [comment]
I have updated a document and now tests are failing. Would this be problematic to merge?

**Comment by pkerpedjiev** [comment]
No, the tests were failing on the develop branch. They should be fixed now though. I'll merge it shortly.

**Pull request merged**